Sources/ChatToys/Utils/WebContext.swift

```diff
@@ -246,38 +246,42 @@ public actor MarkdownProcessor {
 
     private func processWords(_ words: ArraySlice<Substring>) -> String {
         return words.joined(separator: " ")
-            .addingPercentEncoding(withAllowedCharacters: .urlQueryAllowed) ?? ""
             .replacingOccurrences(of: ",", with: "%2C")
             .replacingOccurrences(of: "*", with: "")
             .replacingOccurrences(of: "#", with: "")
             .replacingOccurrences(of: "-", with: "")
             .replacingOccurrences(of: "_", with: "")
             .trimmingCharacters(in: .whitespaces)
+            .addingPercentEncoding(withAllowedCharacters: .urlQueryAllowed) ?? ""
     }
 
     public func shortUUID(_ string: String) -> NodeId {
-        if let existingKey = stringToNodeIdMap[string] {
-            return existingKey
-        }
         var uuid: String
         var collisions = 0
         repeat {
             uuid = String(UUID().uuidString.prefix(uuidLength))
-            if stringToNodeIdMap.values.contains(uuid) {
+            if nodeIdToStringMap.keys.contains(uuid) {
                 collisions += 1
                 if collisions >= 2 {
                     collisions = 0
                     uuidLength += 1
                 }
             }
-        } while stringToNodeIdMap.values.contains(uuid)
-        stringToNodeIdMap[string] = uuid
+        } while nodeIdToStringMap.keys.contains(uuid)
+        nodeIdToStringMap[uuid] = string
         return uuid
     }
 
     public func markdownWithInlineNodeIds(markdown: String, url: String) async -> String {
         var result = [String]()
         for line in markdown.split(separator: "\n") {
+            // Remove any blank list items
+            if line == "-" { continue }
+            // If it's a short text or likely JSON do not give a node ID
+            if line.split(separator: " ").count < 10 || line.prefix(1) == "{" {
```

Comment:
i like the idea of reducing the # of nodes!

i would _imagine_ that if a line doesn't have a node id, the model will assume that the node above it _includes_ it. So it may cite the above node when the info is actually in this one. But in this impl, the node/string map would only include the first line of the node.

So maybe we should:
1. Split the markdown into nodes (deciding whether each new line is worth of a node)
2. After that, insert the whole node into the map.

We could even do fancy stuff like say that a new node isn't created unless the existing node would exceed, say, 150 chars.